sphinx documentation

Kunsi opened the pull request.

Kunsi committed:
start documenting voctomix using sphinx

baldurmen commented:
One thing that would be really nice to have if you're using sphinx is man pages for `voctocore` and `voctogui`. sphinx even makes it easy to embed them in the regular HTML documentation.

fkusei committed:
add Sphinx documentation site with full content migration

- Migrate all existing READMEs into Sphinx RST docs under doc/
- Document all voctocore sources, ports, config options, and CLI flags
- Document all voctogui config options
- Add transitions page with all composite/transition diagrams and GIFs
- Add GitHub Actions workflow to deploy docs to GitHub Pages
- Truncate sub-directory READMEs to redirect links to the docs site
- Configure sphinx-wagtail-theme with voctomix branding and c3voc footer links

fkusei committed:
add man pages for voctocore and voctogui, replace duplicate running docs

Man page RST sources in doc/man/ serve as the single source of truth for
CLI reference — included in the HTML toctree and built as voctocore.1 /
voctogui.1 by the man builder. Removes the separate running.rst and the
inline Running section from voctogui/index.rst.

fkusei committed:
link all source pages to common source attributes

fkusei committed:
remove duplicate voctoremote entry from example scripts

fkusei committed:
document preset buttons and remove dead voctogui config options

The [toolbar] presets list and [preset.<name>] sections control the
quick-switch preset toolbar in voctogui but were entirely missing from
the configuration reference.

Also drop two options from voctogui/default-config.ini that are not
read by any code path (mainwindow/vumeter and previews/use).

fkusei committed:
document preview codec options and clean up dead getters

Adds the previously undocumented [previews] codec options that voctocore
and voctogui actually read (videoencoder, videocodec, videodecoder,
deinterlace, videocaps, audiocaps) and removes documented-but-dead
options (vaapi, scale-method, vaapi-denoise). Also adds [mirrors]/sources
which was missing.

Drops getDenoise and getScaleMethod from vocto/config.py — they have no
callers anywhere in the codebase.

fkusei committed:
document audio.<stream> mapping and clarify volume/preset attribute scope

- Add docs for the audio.<streamname> source attribute (the mechanism
  that exposes named audio streams from source input channels)
- Clarify that the source volume attribute is only read for the blinder
  and document the correct default (1.0)
- Note the auto-fallback for [mix] backgrounds when [source.background]
  exists without an explicit list
- Move icon up to the common toolbar button attributes (the Buttons
  widget supports it for every toolbar section, not just presets) and
  state precisely which common attributes presets honour

fkusei committed:
note that at least one source must define an audio.* stream

voctocore's audio mixer needs at least one input pad — the pipeline
fails to start otherwise.

fkusei committed:
document clock sync port 9998

GStreamer NetTimeProvider on port 9998/UDP was the only port missing
from the reference. Required for any client sending or receiving A/V
streams so its pipeline clock stays in sync with voctocore.

fkusei committed:
fix voctocore -D default and -p man page wording

- Bump -D / --gst-debug-details default from 1 to 15 to match the
  help text (and voctogui's default); update the man page to match.
- Correct the man page description of -p / --pipeline: the pipeline
  is launched in addition to writing the dump, not instead of it.

fkusei committed:
move clock sync port into the central Port constants

The 9998 NetTimeProvider port was hardcoded in voctocore/lib/clock.py
while every other voctomix port lives on the Port class in vocto/port.py.
Centralise it as Port.CLOCK so the constants file is the single source
of truth.

jbeyerstedt commented on voctocore/README.md:
```diff
@@ -1,740 +1 @@
-The real GStreamer pipeline is much more complicated.
-A so-called [DOT graph](https://www.graphviz.org/) of it can be generated by starting **VOC2CORE** with option `-d`. Those DOT graph files can be viewed with [xdot](https://github.com/jrfonseca/xdot.py) for example.
-
-![**VOC2CORE** Mixing Pipeline](images/pipelines.svg)
```
@Kunsi where did this image go in the new documentation? I really like this overview because it shows the general mixing pipeline and network ports in one go.
And in case the diagram is not correct any more, it should be removed. But currently it's just missing from documentation migration.

Kunsi replied:
It wasn't migrated, apparently. It's still in the repo though: https://github.com/voc/voctomix/blob/main/voctocore/images/pipelines.svg

jbeyerstedt replied:
But why wasn't it migrated?
